pkg/profiling/fifo_dir_test.go

```diff
@@ -0,0 +1,52 @@
+package profiling
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_fifoDir(t *testing.T) {
+	dir := t.TempDir()
+	maxFileCount := 3
+	fs := fifoDir{dirPath: dir, maxFileCount: maxFileCount}
+
+	var filesToBeDeleted []string
+	var filesToBeKept []string
+	// This should be bigger than maxFileCount to test FIFO deletion is done properly
+	numFilesToCreate := 10
+	for i := 0; i < numFilesToCreate; i++ {
+		fileName := fmt.Sprintf("%d.test.dump", i)
+		_, err := fs.Create(fileName)
+		require.NoError(t, err, "creating file: %s", fileName)
+		if i < numFilesToCreate-3 {
+			filesToBeDeleted = append(filesToBeDeleted, fileName)
+		} else {
+			filesToBeKept = append(filesToBeKept, fileName)
+		}
+		time.Sleep(time.Millisecond * 100)
+	}
+
+	actualFilesEntries, err := os.ReadDir(dir)
+	require.NoErrorf(t, err, "reading directory %s", dir)
+	actualFiles, err := dirEntriesToFileInfo(actualFilesEntries)
```

Comment:
This function seems to serve both as code under test and test code, which some people might call questionable :wink: